JavaScript/JavaScriptSDK/Logging.ts

```diff
@@ -81,5 +88,57 @@
                 }
             }
         }
+        
+        /**
+         * Resets the internal message count
+         */
+        public static resetInternalMessageCount(): void {
+            this._messageCount = 0;
+        }
+
+        /**
+         * Sets the limit for the number of internal events before they are throttled
+         * @param limit {number} - The throttle limit to set for internal events
+         */
+        public static setMaxInternalMessageLimit(limit: number): void {
+            if (!limit) {
+                throw new Error('limit cannot be undefined.');
+            }
+            
+            this.MAX_INTERNAL_MESSAGE_LIMIT = limit;
+        }
+        
+        /**
+         * Logs a message to the internal queue.
+         * @param severity {LoggingSeverity} - The severity of the log message
+         * @param message {string} - The message to log.
+         */
+        public static logInternalMessage(severity: LoggingSeverity, message: string): void {
+            if (this._areInternalMessagesThrottled()) {
+                return;
+            }
+
+            // Push the event in the internal queue
+            if (this.verboseLogging() || severity === LoggingSeverity.CRITICAL) {
+                if (this.queue.length < this.MAX_QUEUE_SIZE) {
```

Comment:
With `MAX_INTERNAL_MESSAGE_LIMIT` (single digits) being much smaller than the `MAX_QUEUE_SIZE` (triple digits) I wonder if it makes sense to keep both.